### PR TITLE
Refactored to match org docstring style

### PR DESCRIPTION
**Pull request by nukularrr**
* Refactored the docstrings into the numpy/scipy style to better match the MITLL Responsible AI organization style guidelines.

* Added a dark mode option with mkdocs

### CODE REVIEW

**Comment by nukularrr** [comment]
@stevenjson So, Ryan et al. sets a high bar here:
https://github.com/mit-ll-responsible-ai/hydra-zen/blob/main/CONTRIBUTING.md
but I think we can get there. There are a few places that pylance is rightfully complaining in the code, and we "only" have 91% test coverage at the moment. 

Btw, I mostly accomplished the "change docstrings from sphinx format to numpy format" with help from AI -- I did not painfully retype all of our docstrings!

One other thing that is a slight code change is that I made `Protonet` a public class, since folks might actually find that useful, and we had the documentation for it, anyway.

At any rate, take a look at the new output via:
`pip install -e .'[tests,docs]'`
`mkdocs serve`
...or just review the PR and merge it, wait a hot minute, and go to the github pages site. I still need to make something like a github actions test for mkdocs.

**Comment by nukularrr** [comment]
@stevenjson  In this last commit, there are a couple of awkward things I'd like you to review in particular:

1) In adding the FULL covariance type test in EquineProtonet, I had to up the number of shots and eventually add a numpy.abs() before taking the sqrt to avoid getting NaNs in the distances. 
2) It is hard to write a test for `generate_model_summary` since it needs a fully trained model. However, all this is really doing is joining two dictionaries. Unfortunately, the "train_summary" dictionary is added as a member variable to each class after training, which means that downstream users making their own "EquineMyModel" would also have to remember to do that if they wanted to try their own approach. Pylance also rightfully complains that it it can't guarantee the types will match (since there's no guarantee that the dictionary exists). I understand that it's easier to keep the train_summary dict around as metadata / provenance for the last training run, but in that case, we should add it to Equine and initialize all the metrics to 0 (or whatever makes sense) and make it as easy as possible to ensure that the dictionary gets updated after training. 
--OR--
Don't make `train_summar` a member variable and just pass the dict to `generate_model_summary`.

File: src/equine/equine_protonet.py [hunk 1]
```diff
@@ -230,25 +306,43 @@ def compute_distance(
             sol = torch.linalg.lstsq(cov, diff, rcond=10 ** (-4))
             sol = sol.solution  # classes x dimension x examples
             dist = torch.sum(diff * sol, dim=1)  # classes x examples
-            dist = torch.sqrt(dist.permute(1, 0) + self.epsilon)  # examples x classes
+            dist = torch.sqrt(
+                torch.abs(dist.permute(1, 0)) + self.epsilon
```
Comment (stevenjson) [inline src/equine/equine_protonet.py:310]: Just wanted to make sure the abs value code was intened?

**Comment by stevenjson** [comment]
> @stevenjson In this last commit, there are a couple of awkward things I'd like you to review in particular:
> 
> 1. In adding the FULL covariance type test in EquineProtonet, I had to up the number of shots and eventually add a numpy.abs() before taking the sqrt to avoid getting NaNs in the distances.
> 2. It is hard to write a test for `generate_model_summary` since it needs a fully trained model. However, all this is really doing is joining two dictionaries. Unfortunately, the "train_summary" dictionary is added as a member variable to each class after training, which means that downstream users making their own "EquineMyModel" would also have to remember to do that if they wanted to try their own approach. Pylance also rightfully complains that it it can't guarantee the types will match (since there's no guarantee that the dictionary exists). I understand that it's easier to keep the train_summary dict around as metadata / provenance for the last training run, but in that case, we should add it to Equine and initialize all the metrics to 0 (or whatever makes sense) and make it as easy as possible to ensure that the dictionary gets updated after training.
>    --OR--
>    Don't make `train_summar` a member variable and just pass the dict to `generate_model_summary`.

Probably easiest to just initialize the summary when the equine model is created. I can do that quick and push it out.

**Review by stevenjson — APPROVED** [review]
LGTM!

**Pull request merged**